addons/l10n_it_edi_doi/models/account_move.py

```diff
@@ -0,0 +1,238 @@
+    def _compute_l10n_it_edi_doi_warning(self):
+        for move in self:
+            move.l10n_it_edi_doi_warning = ''
+            declaration = move.l10n_it_edi_doi_declaration_of_intent_id
+
+            show_warning = declaration \
+                and move.is_sale_document(include_receipts=False) \
+                and move.state != 'cancel'
+            if not show_warning:
+                continue
+
+            declaration_invoiced = declaration.invoiced
+            declaration_not_yet_invoiced = declaration.not_yet_invoiced
+            if move.state != 'posted':  # exactly the 'posted' invoices are included in declaration.invoiced
+                # Here we replicate what would happen when posting the invoice.
+                # Note: lines manually added to a move linked to an sales order are not added to the sales order
+                declaration_invoiced += move.l10n_it_edi_doi_amount
+                additional_invoiced_qty = {}
+                linked_orders = self.env['sale.order']
+                for invoice_line in move.invoice_line_ids:
+                    for sale_line in invoice_line.sale_line_ids:
+                        order = sale_line.order_id
+                        if order.l10n_it_edi_doi_declaration_of_intent_id == declaration:
+                            linked_orders |= order
+                        qty_invoiced = 0
+                        if invoice_line.move_id.move_type == 'out_invoice':
+                            qty_invoiced += invoice_line.product_uom_id._compute_quantity(invoice_line.quantity, sale_line.product_uom)
+                        elif invoice_line.move_id.move_type == 'out_refund':
+                            qty_invoiced -= invoice_line.product_uom_id._compute_quantity(invoice_line.quantity, sale_line.product_uom)
+                        sale_line_id = sale_line.ids[0]  # do not just use `id` in case of NewId
+                        additional_invoiced_qty[sale_line_id] = additional_invoiced_qty.get(sale_line_id, 0) + qty_invoiced
+                for order in linked_orders:
+                    not_yet_invoiced_before = order._l10n_it_edi_doi_get_amount_not_yet_invoiced(
+                        declaration,
+                    )
+                    not_yet_invoiced = order._l10n_it_edi_doi_get_amount_not_yet_invoiced(
+                        declaration,
+                        additional_invoiced_qty=additional_invoiced_qty,
+                    )
+                    declaration_not_yet_invoiced -= not_yet_invoiced_before - not_yet_invoiced
+
+            validity_warnings = move._l10n_it_edi_doi_get_declaration_of_intent_validity_warnings()
+
+            move.l10n_it_edi_doi_warning = '{}\n\n{}'.format(
+                '\n'.join(validity_warnings),
+                declaration._build_threshold_warning_message(declaration_invoiced, declaration_not_yet_invoiced),
+            ).strip()
```

Comment:
```suggestion
            move.l10n_it_edi_doi_warning = '{}\n\n{}'.format(
                '\n'.join(move._l10n_it_edi_doi_get_declaration_of_intent_validity_warnings()),
                declaration._build_threshold_warning_message(declaration_invoiced, declaration_not_yet_invoiced),
            ).strip()
```
I would either use a variable for both warnings, or none at all :nerd_face: